src/ophyd_async/epics/areadetector/drivers/aravis_driver.py

```diff
@@ -0,0 +1,157 @@
+from enum import Enum
+from math import nan
+from typing import Callable, Dict, Optional, Tuple
+
+from ophyd_async.epics.areadetector.drivers import ADBase
+from ophyd_async.epics.areadetector.utils import ad_r, ad_rw
+
+
+class AravisTriggerMode(str, Enum):
+    """GigEVision GenICAM standard: on=externally triggered"""
+
+    on = "On"
+    off = "Off"
+
+
+class AravisTriggerSource(str, Enum):
+    # While not all enum elements may be physically supported by the hardware,
+    # DB record templates suggest they are valid options for underlying record
+    # cite: https://github.com/areaDetector/ADGenICam/tree/master/GenICamApp/Db
+    # (e.g. Mako G-125B TriggerSource has Line1-4)
+    freerun = "Freerun"
+    fixed_rate = "FixedRate"
+    software = "Software"
+    action_0 = "Action0"
+    action_1 = "Action1"
+    # Externally triggered on GPIO N
+    line_1 = "Line1"
+    line_2 = "Line2"
+    line_3 = "Line3"
+    line_4 = "Line4"
+
+
+def _reverse_lookup(
+    model_deadtimes: Dict[float, Tuple[str, ...]]
+) -> Callable[[str], float]:
+    def inner(pixel_format: str) -> float:
+        for deadtime, pixel_formats in model_deadtimes.items():
+            if pixel_format in pixel_formats:
+                return deadtime
+        return nan
```

Comment:
I think I'd probably prefer to `raise ValueError` or something similar rather than return a `NaN` here.

The reason I prefer this is that `NaN` propagates quietly through maths operations and then is more likely to show an obscure problem later on, while really what I want to know is that it got an invalid/unknown pixel format.

Please can we also write a unit test for the invalid/unknown pixel format case.